File: src/main/java/com/example/cqrs/command/dto/TransferRequest.java
package com.example.cqrs.command.dto;

import lombok.*;

@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
@Getter
public class TransferRequest {

    private String fromAccountId;
    private String toAccountId;
    private double amount;
    private String userId;

    // factory method
    public static TransferRequest of(String fromAccountId, String toAccountId, double amount, String userId) {
        return new TransferRequest(fromAccountId, toAccountId, amount, userId);
    }

}

File: src/main/java/com/example/cqrs/command/dto/TransferRequest.kt
package com.example.cqrs.command.dto

data class TransferRequest(
    val fromAccountId: String,
    val toAccountId: String,
    val amount: Double,
    val userId: String
) {
    companion object {
        fun of(fromAccountId: String, toAccountId: String, amount: Double, userId: String): TransferRequest {
            return TransferRequest(fromAccountId, toAccountId, amount, userId)
        }
    }
}
